### DISCUSSION TITLE
Review halstack-react library dependencies and compatibilities

### DISCUSSION THREAD

**Discussion by GomezIvann** [dxc-technology/halstack-react#1664] [Ideas] [answered]
Halstack has been stacked in npm <= 6 and Node 14 for a while. Our dependencies keep giving problems to projects and users with newer versions of those technologies. This is a big issue since Halstack relies on not too many dependencies (mainly react and styled-components) to operate and should be straightforward.

We need to review our whole package.json file to have a more logical distribution of our dependencies as a library of react components:

- `devDependencies`: keep only the ones we need. We may have too many coming from `create-react-app` clouding our library.
- `dependencies`: we have too many of them installed in our users' projects with halstack and we do not know for sure if they need them. Some may go on `devDependencies` instead.
- `peerDependencies`: I'm pretty confident that the only dependencies we need for `@dxc-technology/halstack-react` to work fine are the ones we have. However, we can still review them.

Some additional links for further knowledge and clarification:

1. [Difference between dependencies, devDependencies and peerDependencies](https://www.geeksforgeeks.org/difference-between-dependencies-devdependencies-and-peerdependencies/)
2. https://stackoverflow.com/questions/18875674/whats-the-difference-between-dependencies-devdependencies-and-peerdependencie
3. [Peer dependencies in npm 7 are installed automatically](https://github.blog/2021-02-02-npm-7-is-now-generally-available/#peer-dependencies)

**Comment by raquelarrojo** [comment]
I totally agree. I would left only the essential ones in `dependencies` of our library, like `react` and some others.

With the current configuration we have, if we install node 15 (that includes npm 7) we obtain an error due to incompatibilities in our dependencies:

<img width="721" alt="image" src="https://github.com/dxc-technology/halstack-react/assets/10975289/da878def-3045-42fd-be58-c19dedcb6135">


We should fix this since, as @GomezIvann said, it should be straightforward to use our Halstack library, and we could be interested on upgrade our npm version to 7 since it automatically installs the peerDependencies.

**Reply by kukoc** [reply, marked as answer]
Let's target Node 16 and npm 7 and upgrade our package files and workflows. During the process, let's also review the dependencies of the Library and double-check if they're needed.
